CMakeLists.txt

```diff
@@ -641,7 +641,12 @@ endif()
 add_library(avif ${AVIF_SRCS})
 set_target_properties(avif PROPERTIES VERSION ${LIBRARY_VERSION} SOVERSION ${LIBRARY_SOVERSION} C_VISIBILITY_PRESET hidden)
 target_compile_definitions(avif PRIVATE ${AVIF_PLATFORM_DEFINITIONS} ${AVIF_CODEC_DEFINITIONS})
-target_link_libraries(avif PRIVATE ${AVIF_CODEC_LIBRARIES} ${AVIF_PLATFORM_LIBRARIES})
+if(BUILD_SHARED_LIBS)
+    target_link_libraries(avif PRIVATE ${AVIF_CODEC_LIBRARIES} ${AVIF_PLATFORM_LIBRARIES})
+else()
+    include(merge_static_libs)
+    merge_static_libs(avif ${AVIF_CODEC_LIBRARIES} ${AVIF_PLATFORM_LIBRARIES})
```

Comment:
I didn't know that the decision to merge static libraries does not take the `AVIF_LOCAL_AOM`, `AVIF_LOCAL_DAV1D`, ... variables into account. I thought we would only merge libavif.a with the libaom, dav1d, ... static libraries when they were built locally in the ext/ directory.

If the libaom, dav1d, ... static libraries are installed as system packages (I don't know if system packages provide static libraries, but for this discussion let's assume they do) and `AVIF_LOCAL_AOM`, `AVIF_LOCAL_DAV1D` are equal to `OFF`, it doesn't seem appropriate to merge libavif.a with the static libraries installed on the system.

I am wondering if we should abandon the merge_static_libs approach and seek an alternative solution to the problem that Frankie needed to solve.

Reply:
I'm not sure the behavior should be different for system static libraries vs local libraries. In either case the static library isn't really portable; whoever is linking against the library would need to know which codecs it was linked against and have ABI compatible static libraries installed on that system. And often times the build environment happens in CI, typically on a different platform than where it eventually gets used.

I think the "proper" way to do this is to include the statically-linked system libraries in Libs.private of libavif.pc. This would allow (for instance) another cmake project to call `pkg_check_modules` to get the necessary linker flags. But I'm not sure that's superior to creating a combined archive.